app/routes/circulars._archive._index/SynonymGroupIndex.tsx

```diff
@@ -5,17 +5,70 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-import { Link } from '@remix-run/react'
+import { Link, useFetcher } from '@remix-run/react'
+import type { action } from 'app/routes/synonyms/route'
+import { useEffect, useState } from 'react'
 
-import type { SynonymGroupWithMembers } from '../synonyms/synonyms.lib'
+import type { Circular } from '../circulars/circulars.lib'
+import type { SynonymGroup } from '../synonyms/synonyms.lib'
+import Spinner from '~/components/Spinner'
+
+function SynonymList({
+  eventIds,
+  slugs,
+  searchString,
+}: {
+  eventIds: string[]
+  slugs: string[]
+  searchString: string
+}) {
+  const [items, setItems] = useState<Circular[]>([])
+  const fetcher = useFetcher<typeof action>()
+  const data = new FormData()
+  data.set('eventIds', eventIds.join(','))
+
+  useEffect(() => {
+    setItems(fetcher.data ?? [])
+  }, [fetcher.data])
```

Comment:
Why do you need the state and effect hooks? Why can't you just use `fetcher.data` in your component code below?